### DISCUSSION TITLE
Question about the number of 02.fp tasks (task.00x.xxxxx)

### DISCUSSION THREAD

**Discussion by nzynpu** [deepmodeling/dpgen#934] [Q&A] [answered]
Without changing param.json, changing the number of "sys_configs"(from 3 to 24) got different number of tasks in 02.fp, my understanding is that the number of tasks in 02.fp depends on the choice of 01.model_devi parameter, what can I do except change the number of “sys_configs", can the purpose of changing the number of tasks in 02.fp be achieved by changing the parameters of 01.model_devi, and if so, what parameters need to be changed? Thanks very much.

**Comment by HuangJiameng** [comment, marked as answer]
The task number in `02.fp` is minimum of the number of candidates that picked in `01.model_devi` step and the value of `fp_task_max`. It has nothing to do with `sys_configs`. see https://docs.deepmodeling.com/projects/dpgen/en/latest/run/param.html#argument:run_jdata/fp_task_max

**Reply by njzjz** [reply]
Indeed `fp_task_max` is the maximum of structures **in each system**. The documentation needs to be more precise.

**Reply by nzynpu** [reply]
Thanks for your reply, I get it. In my training, the number of 01.model_devi candidates is less than the value of fp_task_max, the number of 01.model_devi candidates determines the number of 02.fp tasks, so how can I increase the number of candidates?

**Reply by HuangJiameng** [reply]
If the number of candidates is less than `fp_task_max`, you can increase its value. Or you can change the value of [model_devi_f_trust_lo](https://docs.deepmodeling.com/projects/dpgen/en/latest/run/param.html#argument:run_jdata[model_devi_engine=lammps]/model_devi_jobs/model_devi_f_trust_lo) and [model_devi_f_trust_hi](https://docs.deepmodeling.com/projects/dpgen/en/latest/run/param.html#argument:run_jdata[model_devi_engine=lammps]/model_devi_jobs/model_devi_f_trust_hi), according to which candidates are picked.

**Reply by nzynpu** [reply]
Thanks very much,i solved this question.
